### DISCUSSION TITLE
large collection of videos

### DISCUSSION THREAD

**Discussion by jindal1979** [unoplatform/uno#12114] [Q&A]
I want to display an extensive collection of videos with its thumbnail in List and Grid Mode. What is the preferred way to display it?

I am currently using ListView, however, the performance is too slow on Android. Please advise, is there any sample available that displays a large collection without performance issues?

**Comment by FrozDark** [comment]
Android is poor performant

**Reply by jindal1979** [reply]
What is the solution?

**Reply by jeromelaban** [reply]
What do you mean by performance is too slow? Do you observe the performance issue in Release mode as well?

**Reply by FrozDark** [reply]
> What do you mean by performance is too slow? Do you observe the performance issue in Release mode as well?

This is the demonstration of Uno Android on 4.9.0-dev.739 in Release
https://www.veed.io/view/aa98d5c2-a655-466b-a4fd-6ae64e957750

There is also a big app size (Mine is ~77 MB in release). There might be worth playing with linker but when I had been trying to I always failed so I've used default settings

Device: Galaxy A11
OS: Android 12
Core: Qualcomm Snapdragon 450 1,8GHz (8 core)
RAM: 2 GB

**Reply by jeromelaban** [reply]
The video is not publicly accessible, it's probably best if you attach it here.

**Reply by FrozDark** [reply]
> The video is not publicly accessible, it's probably best if you attach it here.

[2023_05_01_22_25_33.zip](https://github.com/unoplatform/uno/files/11369747/2023_05_01_22_25_33.zip)

**Reply by jindal1979** [reply]
> What do you mean by performance is too slow? Do you observe the performance issue in Release mode as well?

I am displaying 150 elements in ListView. It takes too much time to load ListView (even in Release mode). However, if I use ItemsRepeater, it loads it pretty quickly. So, there are definitely performance issues in ListView.
